catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java

```diff
@@ -155,6 +157,46 @@ public void init() {
         });
   }
 
+  @Override
+  public Object apply(Object req) {
+    if (!(req instanceof List)) {
+      return JsonRpc.invalidParams("params not list", req);
+    }
+
+    List params = (List) req;
+
+    if (params.size() != 1) {
+      return JsonRpc.invalidParams("must pass exactly 1 param", params);
+    }
+
+    Object param = params.get(0);
+
+    if (!(param instanceof String)) {
+      return JsonRpc.invalidParams("param not string", param);
+    }
+
+    CqlRequest cqlRequest;
+
+    try {
+      cqlRequest = mapper.readValue((String) param, CqlRequest.class);
+    } catch (RuntimeException e) {
```

Comment:
Can this be a more specific exception? Sonarqube is going to complain about this. If it needs to be RuntimeException then can you add the suppression for it to the method.

Reply:
I am implementing the same behavior as the query endpoint here, so I don't want to change how exceptions map to error messages.